pkg/cmd/init.go

```diff
@@ -346,6 +345,33 @@ func initKubeconfigs(
 	if err != nil {
 		return err
 	}
+
+	u, err := url.Parse(cfg.Cluster.URL)
+	if err != nil {
+		return fmt.Errorf("failed to parse cluster URL: %v", err)
+	}
+	apiServerPort, err := cfg.Cluster.ApiServerPort()
+	if err != nil {
+		return fmt.Errorf("failed to get apiserver port: %v", err)
+	}
+
+	// If using localhost this is part of the default URL and it needs external access.
+	// Generate all other kubeconfigs, one per name.
+	if u.Hostname() == "localhost" || u.Hostname() == "127.0.0.1" {
+		for _, name := range append(cfg.SubjectAltNames, cfg.NodeName, "localhost") {
```

Comment:
Ok, to fully end the issue we also need another patch to split NodeName & HostName

But that makes me think, what will happen with URL, which contains a hostname.

Should we remove the URL parameter?, and construct URL From protocol + hostname + port ?

Reply:
Or should URL be something different that will simply differ across kubeconfig files?

This is the only part were I believe we need more consistence and agreement, but it's not specific to this patch.

Reply:
> Ok, to fully end the issue we also need another patch to split NodeName & HostName
> 
> But that makes me think, what will happen with URL, which contains a hostname.
> 
> Should we remove the URL parameter?, and construct URL From protocol + hostname + port ?

It is going to have a different [name](https://github.com/openshift/microshift/pull/1030/files#diff-e279d89e0daa9e6b083e46b5c0f2b9d18015b12fb264fd7caf6fb3a5d4c3ee52R37) but it serves the same purpose, as a URL. I have to say I see the URL more as a hostname+port thing (or even only port, having the hostname/nodename and SAN already there). Is there any use case where the apiserver is going to require a path in that URL?



> Or should URL be something different that will simply differ across kubeconfig files?
> 
> This is the only part were I believe we need more consistence and agreement, but it's not specific to this patch.

Your first sentence is what this PR is doing, generate a set of kubeconfigs only with different `server` value.

Reply:
Sorry for my previous comments; I was trying to grasp the whole picture.

I believe the proposed approach makes sense. The Hostname is derived from the URL instead of NodeName, so we already have the two different parameters we wanted, and there is no need for more config parameters.